pytorch_lightning/plugins/training_type/training_type_plugin.py

```diff
@@ -246,6 +246,14 @@ def restore_checkpoint_after_pre_dispatch(self) -> bool:
         """
         return False
 
+    @property
+    def lightning_restore_optimizer_and_schedulers(self) -> bool:
```

Comment:
does this really has to be a property on the instance level? wouldn't a static attribute be fine as well? I don't expect that to change per instance of the plugin, right?

Reply:
Yep definitely can be an attribute, would that be prefered?

Reply:
From our offline sync (just for completeness): 
> No strong opinion, they both have advantages :) Class attributes are the same for all instances of the class (which they should be IMO) but properties would fail when overwriting (if not explicitly defined a setter) which also makes sense here

Reply:
I'll keep it as is, but this was a good lesson for me as I never considered the behaviours of class attribute vs property like this before :) 

If anyone has a strong preference either way, I can change this.